src/IO/H5/CheckH5PropertiesMatch.cpp

```diff
@@ -0,0 +1,82 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "IO/H5/CheckH5PropertiesMatch.hpp"
+
+#include <cstddef>
+#include <iterator>
```

Comment:
include `<vector>, <string>` I'm not seeing what `<iterator>` is being used for in this file. 

Note: we don't have a solid policy as to whether to include a file "again" in a cpp file. I'm not self-consistent either, so up to you!